docs/design/deviceshifu/design-deviceShifu-rtsp.md

```diff
@@ -0,0 +1,93 @@
+# deviceShifu Rtsp overall design
+
+deviceShifu-RTSP allows shifu connect to rtsp server and save the vedio into file. User can watch rtsp vedio using media player like VLC and get history vedio file from PV.
+
+## Goal
+
+Create a deviceShifu-RTSP using thired-party library and allow user to connect RTSP server, and record the rtsp stream into file like mp4 / flv and so on, user can export file out of k8s.
+
+## General Design
+
+deviceShifu-RTSP will receive RESTful style request like other, then transfer the request into RTSP method calls and send it to RTSP Server, meanwhile deviceShifu-RTSP provide a RTSP address which forward the RTSP stream from the target RTSP server.
+
+## Detailed Design
+
+### Protocol Specification
+
+deviceShifu-RTSP provide a RTSP server to forward RTSP stream and HTTP Server to handle Request.
+
+```go
+type ProtocolSettings struct {
+    MQTTSetting *MQTTSetting `json:"MQTTSetting,omitempty"`
+    ....
++   RTSPSetting *RTSPSetting `json:"RTSPSetting,omitempty"`
+}
+
+type RTSPSetting struct {
+    Username        string `json:"username,omitempty"`
+    PasswordSecret  string `json:"passwordSecret,omitempty"`
+    Record          bool   `json:"record,omitempty"`
```

Comment:
the username/password is used only if `Record` is `true` right?

Reply:
for the RTSP URL like `rtsp://username:password@ip ...` so the username and password will be used in every Functionality

Reply:
but if we forward the connection, how does the user use the localhost stream in tools like VLC:
e.g. `rtsp://{username}:{password}@localhost:{port}` ?